### ISSUE TITLE
Issue on the commit fb67978a9* on OpenSSL_1_1_1-stable branch: OpenSSL Build Error

### ISSUE THREAD

**Issue by henghengxqc** [openssl/openssl#17999]
git command: 
> git clone --recurse-submodules -b OpenSSL_1_1_1-stable https://github.com/openssl/openssl

Config option: 
> ./config --prefix=/home/openssl/target --openssldir=/home/openssl/target shared -d

Build command: 
> make -j4

Error: 
```
gcc  -I. -Iinclude -fPIC -pthread -m64 -Wa,--noexecstack -Wall -O0 -g -DOPENSSL_USE_NODELETE -DL_ENDIAN -DOPENSSL_PIC -DOPENSSL_CPUID_OBJ -DOPENSSL_IA32_SSE2 -DOPENSSL_BN_ASM_MONT -DOPENSSL_BN_ASM_MONT5 -DOPENSSL_BN_ASM_GF2m -DSHA1_ASM -DSHA256_ASM -DSHA512_ASM -DKECCAK1600_ASM -DRC4_ASM -DMD5_ASM -DAESNI_ASM -DVPAES_ASM -DGHASH_ASM -DECP_NISTZ256_ASM -DX25519_ASM -DPOLY1305_ASM -DOPENSSLDIR="\"/home/openssl/target\"" -DENGINESDIR="\"/home/openssl/target/lib/engines-1.1\""   -MMD -MF ssl/statem/statem_lib.d.tmp -MT ssl/statem/statem_lib.o -c -o ssl/statem/statem_lib.o ssl/statem/statem_lib.c
ssl/statem/statem_clnt.c: In function 'tls_process_server_hello':
ssl/statem/statem_clnt.c:1426:76: error: macro "SSLfatal" requires 4 arguments, but only 3 given
             SSLfatal(s, SSL_AD_UNEXPECTED_MESSAGE, SSL_R_UNEXPECTED_MESSAGE);
                                                                            ^
ssl/statem/statem_clnt.c:1426:13: error: 'SSLfatal' undeclared (first use in this function); did you mean 'SSL_ctrl'?
             SSLfatal(s, SSL_AD_UNEXPECTED_MESSAGE, SSL_R_UNEXPECTED_MESSAGE);
             ^~~~~~~~
             SSL_ctrl
ssl/statem/statem_clnt.c:1426:13: note: each undeclared identifier is reported only once for each function it appears in
Makefile:6220: recipe for target 'ssl/statem/statem_clnt.o' failed
make[1]: *** [ssl/statem/statem_clnt.o] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: Leaving directory '/home/openssl'
Makefile:174: recipe for target 'all' failed
make: *** [all] Error 2
```

-----

When I reset the commit to the previous one, the error was gone.
> git reset --hard 739d2bdfba536ff59e8444eb4295b53288ac5caf

The fix:
Change `ssl/statem/statem_clnt.c:1426` from:
```
SSLfatal(s, SSL_AD_UNEXPECTED_MESSAGE, SSL_R_UNEXPECTED_MESSAGE);
```
to:
```
SSLfatal(s, SSL_AD_UNEXPECTED_MESSAGE, SSL_R_UNEXPECTED_MESSAGE, SSL_R_LENGTH_MISMATCH);
```

**Comment by henghengxqc** [comment]
@t8m FYI

**Comment by mattcaswell** [comment]
Fix in #18000
